go/vt/vtgate/vindexes/numeric_static_map_hash_test.go

```diff
@@ -0,0 +1,131 @@
+/*
+Copyright 2017 Google Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package vindexes
+
+import (
+	"reflect"
+	"testing"
+
+	"strings"
+
+	"github.com/youtube/vitess/go/sqltypes"
+	"github.com/youtube/vitess/go/testfiles"
+)
+
+// createNumericStaticMapHashVindex creates the "numeric_static_map" vindex object which is used by
+// each test.
+//
+// IMPORTANT: This code is called per test and must not be called from init()
+// because our internal implementation of testfiles.Locate() does not support to
+// be called from init().
+func createNumericStaticMapHashVindex() (Vindex, error) {
+	m := make(map[string]string)
+	m["json_path"] = testfiles.Locate("vtgate/numeric_static_map_hash_test.json")
+	return CreateVindex("numeric_static_map_hash", "NumericStaticMapHash", m)
+}
+
+func TestNumericStaticMapHashCost(t *testing.T) {
+	NumericStaticMapHash, err := createNumericStaticMapHashVindex()
+	if err != nil {
+		t.Fatalf("failed to create vindex: %v", err)
+	}
+	if NumericStaticMapHash.Cost() != 1 {
+		t.Errorf("Cost(): %d, want 1", NumericStaticMapHash.Cost())
+	}
+}
+
+func TestNumericStaticMapHashString(t *testing.T) {
+	NumericStaticMapHash, err := createNumericStaticMapHashVindex()
+	if err != nil {
+		t.Fatalf("failed to create vindex: %v", err)
+	}
+	if strings.Compare("NumericStaticMapHash", NumericStaticMapHash.String()) != 0 {
+		t.Errorf("String(): %s, want num", NumericStaticMapHash.String())
+	}
+}
+
+func TestNumericStaticMapHashMap(t *testing.T) {
+	NumericStaticMapHash, err := createNumericStaticMapHashVindex()
+	if err != nil {
+		t.Fatalf("failed to create vindex: %v", err)
+	}
+	got, err := NumericStaticMapHash.(Unique).Map(nil, []sqltypes.Value{
+		sqltypes.NewInt64(1),
+		sqltypes.NewInt64(2),
+		sqltypes.NewInt64(3),
+		sqltypes.NewInt64(4),
+		sqltypes.NewInt64(5),
+		sqltypes.NewInt64(6),
+		sqltypes.NewInt64(7),
+		sqltypes.NewInt64(8),
+	})
+	if err != nil {
+		t.Error(err)
+	}
+
+	// in the third slice, we expect 2 instead of 3 as numeric_static_map_test.json
+	// has 3 mapped to 2
+	want := [][]byte{
+		[]byte("\x00\x00\x00\x00\x00\x00\x00\x01"),
```

Comment:
Except for 1 & 3, the outputs should be hashed. Look at https://github.com/youtube/vitess/blob/master/go/vt/vtgate/vindexes/hash_test.go#L61 for examples.

Reply:
Fixed